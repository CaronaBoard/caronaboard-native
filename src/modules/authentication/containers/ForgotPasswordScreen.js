import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { forgotPasswordFirebase } from '../../../redux/actions'
import { RkText, RkTextInput, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { styles } from '../containers/styles/LoginScreenStyles'
import { View, Image, Keyboard } from 'react-native'
import { GradientButton } from '../../shared/components/index'
import Toast, { DURATION } from 'react-native-easy-toast'

export class ForgotPasswordScreen extends Component {
  static defaultProps = {
    toast: {
      showAlert: false,
      message: ''
    }
  }

  static propTypes = {
    resetPassword: PropTypes.func.isRequired,
    alert: PropTypes.shape({
      showAlert: PropTypes.bool.isRequired,
      message: PropTypes.string
    }).isRequired
  }

  state = {
    email: '',
    alert: {
      showAlert: false,
      message: ''
    }
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.alert

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
      setTimeout(() => {
        this.props.navigator.push({
          screen: 'carona.signIn',
          title: 'Sign In Screen'
        })
      }, DURATION.LENGTH_LONG)
    }
  }

  render () {
    const { resetPassword } = this.props
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/caronaboard-logo-azul.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <RkTextInput
          onChangeText={(email) => { this.setState({email}) }}
          key='email-input'
          rkType='rounded'
          placeholder='Username'
          autoCapitalize='none'
        />
        <View style={styles.centralized}>
          <GradientButton
            rkType='large'
            style={styles.loginButton}
            text='Send'
            onPress={() => resetPassword(this.state.email)}
          />
        </View>
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    alert: state.auth.alert
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    resetPassword: (email) => dispatch(forgotPasswordFirebase(email))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ForgotPasswordScreen)
