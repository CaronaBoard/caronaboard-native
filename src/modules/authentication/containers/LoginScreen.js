import React from 'react'
import {
  View,
  Image,
  Keyboard
} from 'react-native'
import {
  RkButton,
  RkText,
  RkTextInput,
  RkAvoidKeyboard,
  RkTheme
} from 'react-native-ui-kitten'
import { GradientButton } from '../../shared/components'
import { styles } from './styles/LoginScreenStyles'

export default class LoginV2 extends React.Component {
  onLoginButtonPress = () => {
  }

  navigateToSignUpScreen = () => {
  }

  renderIcon = () => {
    let source
    if (RkTheme.current.name === 'light') {
      source = require('../../../assets/images/logo.png')
    } else {
      source = require('../../../assets/images/logoDark.png')
    }

    return <Image style={styles.image} source={source} />
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          {this.renderIcon()}
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <RkTextInput rkType='rounded' placeholder='Username' />
        <RkTextInput rkType='rounded' placeholder='Password' secureTextEntry />

        <View style={styles.centralized}>
          <GradientButton style={styles.loginButton} rkType='large' text='LOGIN' onPress={this.onLoginButtonPress} />
        </View>

        <View style={styles.textRow}>
          <RkText rkType='primary3'>Donâ€™t have an account?</RkText>
          <RkButton rkType='clear' onPress={this.navigateToSignUpScreen}>
            <RkText rkType='header6'> Sign up now </RkText>
          </RkButton>
        </View>

      </RkAvoidKeyboard>
    )
  }
}
