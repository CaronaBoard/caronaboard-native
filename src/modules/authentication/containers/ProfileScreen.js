import Styles from './styles/ProfileScreenStyles'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { View, TouchableOpacity } from 'react-native'
import { RkText, RkChoice, RkChoiceGroup } from 'react-native-ui-kitten'
import type { ProfileType } from '../../../services/firebase/types'
import { ProfilePropType } from '../types'

import { TextInput, GradientButton } from '../../shared/components'
import { saveProfileFirebase } from '../../../redux/actions'

export const CONTACT_OPTIONS = ['Whatsapp', 'Telegram']
export const INITIAL_STATE: ProfileType = {
  profile: {
    name: '',
    uid: '',
    contact: {
      kind: '',
      value: ''
    }
  },
  loading: false
}

export class ProfileScreen extends Component {
  static propTypes = {
    navigator: PropTypes.object.isRequired,
    userId: PropTypes.string.isRequired,
    profile: ProfilePropType
  }

  state = INITIAL_STATE

  onContactOptionSelected = kind => {
    const { profile } = this.state
    let { contact } = profile
    contact.kind = kind
    this.setState({profile: {...profile, contact}})
  }

  onButtonSubmit = async () => {
    this.setState({loading: true})

    try {
      this.props.saveProfile(this.state.profile)
    } catch (error) {
      console.error(error)
    }

    this.setState({loading: false})
  }

  renderOption = (option, i) => (
    <View key={i}>
      <TouchableOpacity choiceTrigger>
        <View style={Styles.contactOption}>
          <RkChoice rkType='radio' />
          <RkText>{option}</RkText>
        </View>
      </TouchableOpacity>
    </View>
  )

  componentDidMount () {
    const { profile } = this.props
    this.setState({profile, loading: false})
  }

  render () {
    const { profile } = this.state
    const { contact, name } = profile

    return (
      <View style={Styles.flexible}>
        <View style={Styles.container}>
          <RkText style={Styles.title}>
            PROFILE
          </RkText>
          <View style={Styles.inputTextsContainer}>
            <TextInput
              placeholder='Your name'
              value={name}
              onChangeText={name => { this.setState({profile: {...profile, name}}) }}
            />
          </View>
          <RkChoiceGroup
            radio
            rkType='clear'
            onChange={index => this.onContactOptionSelected(CONTACT_OPTIONS[index])}>
            {CONTACT_OPTIONS.map(this.renderOption)}
          </RkChoiceGroup>
          <View style={Styles.inputTextsContainer}>
            <TextInput
              placeholder='Number'
              value={contact.value}
              onChangeText={value => { this.setState({profile: {...profile, contact: {...contact, value}}}) }}
            />
          </View>
        </View>
        <View style={Styles.centralized}>
          <GradientButton
            rkType='large'
            text={'SAVE'}
            onPress={this.onButtonSubmit}
          />
        </View>
      </View>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    profile: state.auth.profile, // TODO: do we have profile saved into state?
    userId: state.auth.userData.uid
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    saveProfile: (profile, userId) => dispatch(saveProfileFirebase(profile, userId))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ProfileScreen)
