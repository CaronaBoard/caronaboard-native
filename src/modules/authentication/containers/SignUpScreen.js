import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { signUpFirebase } from '../../../redux/actions'
import { LoginForm } from '../components/LoginForm'

export class SignUpScreen extends Component {
  static propTypes = {
    signUp: PropTypes.func.isRequired,
    alert: PropTypes.shape({
      showAlert: PropTypes.bool.isRequired,
      message: PropTypes.string
    }).isRequired
  }

  render () {
    return (
      <LoginForm
        onButtonPress={this.props.signUp}
        toast={this.props.alert}
        buttonText='SIGN UP'
      />
    )
  }
}

const mapStateToProps = (state) => {
  return {
    alert: state.auth.alert
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    signUp: (email, password) => dispatch(signUpFirebase(email, password))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SignUpScreen)
