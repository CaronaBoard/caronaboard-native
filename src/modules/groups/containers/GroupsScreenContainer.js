import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { GroupsScreen } from '../components/GroupsScreen'
import { askToJoinGroup, changeGroup, fetchGroups } from '../../../services/firebase/database/Groups'
import { LoadingSpinnerView } from '../../shared/components/LoadingSpinnerView'
import { confirmationAlert } from '../../../navigation/Alert'
import { updateRideGroup } from '../../../redux/actions/sync/RideRequestActions'

class GroupsScreenContainer extends Component {
  static propTypes = {
    navigator: PropTypes.object.isRequired,
    profile: PropTypes.object.isRequired,
    group: PropTypes.object.isRequired,
    updateGroup: PropTypes.func.isRequired
  }

  state = {
    groups: [],
    loading: false
  }

  async componentDidMount () {
    const groups = await fetchGroups()
    this.setState({groups, loading: false})
  }

  joinGroup = async (group) => {
    const { state } = this
    this.setState({...state, loading: true})
    confirmationAlert('Deseja solicitar entrada para este grupo?', async () => askToJoinGroup(group, this.props.profile))
    this.setState({...state, loading: false})
  }

  selectGroup = async (group) => {
    this.props.updateGroup(group)
    changeGroup(group.id)
  }

  render () {
    return (
      <LoadingSpinnerView isLoading={this.state.loading}>
        <GroupsScreen
          groups={this.state.groups}
          uid={this.props.profile.uid}
          joinGroup={this.joinGroup}
          changeGroup={this.selectGroup}
          activeGroup={this.props.group}
        />
      </LoadingSpinnerView>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    profile: state.auth.profile,
    group: state.ride.group
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    updateGroup: group => (dispatch(updateRideGroup(group)))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(GroupsScreenContainer)
