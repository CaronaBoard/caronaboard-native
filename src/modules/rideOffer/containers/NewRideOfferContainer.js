import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { onNavigatorEvent } from '../../../navigation/NavBar'
import { saveRideOffer } from '../../../services/firebase'
import { NewRideOffer } from '../components/NewRideOffer'
import { fetchYourRideOffers } from '../../../redux/actions/async/RideOfferActions'

export class NewRideOfferContainer extends Component {
  static propTypes = {
    navigator: PropTypes.object.isRequired,
    updateYourOffers: PropTypes.func.isRequired,
    userId: PropTypes.string.isRequired
  }

  constructor (props) {
    super(props)
    this.props.navigator.setOnNavigatorEvent(onNavigatorEvent.bind(this))
  }

  offerRide = async (rideOffer) => {
    const { updateYourOffers, userId } = this.props
    try {
      await saveRideOffer(rideOffer, this.props.userId)
      await updateYourOffers(userId)
      alert('Success')
    } catch (error) {
      alert('Error')
    }
  }

  render () {
    return <NewRideOffer onPress={this.offerRide} />
  }
}

const mapStateToProps = (state) => {
  return {
    userId: state.auth.userData.uid
  }
}

const mapDispatchToProps = dispatch => {
  return {
    updateYourOffers: uid => dispatch(fetchYourRideOffers(uid))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(NewRideOfferContainer)
