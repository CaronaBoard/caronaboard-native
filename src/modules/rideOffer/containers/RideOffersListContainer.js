import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { ListView } from 'react-native'
import React, { Component } from 'react'

import { RideOffer } from '../components/RideOffer'
import { screens } from '../../../navigation/Screens'
import { RidePropType } from '../../rideRequest/types'
import { fetchAllRideOffers } from '../../../redux/actions'
import { onNavigatorEvent } from '../../../navigation/NavBar'
import { LoadingSpinnerView } from '../../shared/components/LoadingSpinnerView'

export class RideList extends Component {
  static propTypes = {
    navigator: PropTypes.object.isRequired,
    rides: PropTypes.arrayOf(RidePropType).isRequired,
    userId: PropTypes.string.isRequired
  }

  state = {
    loading: false
  }

  constructor (props) {
    super(props)
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1.id !== r2.id})
    this.props.navigator.setOnNavigatorEvent(onNavigatorEvent.bind(this))
    this.state = {dataSource: ds.cloneWithRows(this.props.rides)}
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.rides.length !== this.props.rides.length) {
      const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1.id !== r2.id})
      this.setState({dataSource: ds.cloneWithRows(nextProps.rides)})
    }
  }

  async componentDidMount () {
    const { userId, fetchRides } = this.props
    if (userId) {
      this.setState({loading: true})
      await fetchRides()
      this.setState({loading: false})
    }
  }

  async componentDidUpdate (prevProps) {
    const { userId } = this.props
    if (userId && prevProps.userId !== userId) {
      await this.componentDidMount()
    }
  }

  onPress = (ride) => {
    const { navigator } = this.props
    navigator.push({
      screen: screens.rideRequest.id,
      title: screens.rideRequest.title,
      passProps: {ride}
    })
  }

  render () {
    return (
      <LoadingSpinnerView isLoading={this.state.loading}>
        <ListView
          dataSource={this.state.dataSource}
          renderRow={(ride) => <RideOffer ride={ride} onPress={this.onPress} />}
          enableEmptySections
        />
      </LoadingSpinnerView>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    rides: state.ride.offers,
    userId: state.auth.userData.uid
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchRides: () => dispatch(fetchAllRideOffers())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(RideList)
