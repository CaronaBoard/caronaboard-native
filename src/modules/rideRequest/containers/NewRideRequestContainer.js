import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { NewRideRequest } from '../components/NewRideRequest'
import { RidePropType } from '../types'
import { saveRideRequest } from '../../../services/firebase'
import { fetchYourRideRequests } from '../../../redux/actions/async/RideRequestActions'

export class RideRequestScreen extends Component {
  static propTypes = {
    ride: RidePropType.isRequired,
    updateYourRequests: PropTypes.func.isRequired,
    userId: PropTypes.string.isRequired
  }

  newRideRequest = async (rideId: string) => {
    const { updateYourRequests, userId } = this.props
    try {
      await saveRideRequest(rideId, this.props.userId)
      await updateYourRequests(userId)
      alert('Success')
    } catch (error) {
      alert('Error')
    }
  }

  render () {
    return (
      <NewRideRequest
        ride={this.props.ride}
        saveRideRequest={this.newRideRequest}
      />
    )
  }
}

const mapStateToProps = (state) => {
  return {
    userId: state.auth.userData.uid
  }
}

const mapDispatchToProps = dispatch => {
  return {
    updateYourRequests: uid => dispatch(fetchYourRideRequests(uid))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(RideRequestScreen)
