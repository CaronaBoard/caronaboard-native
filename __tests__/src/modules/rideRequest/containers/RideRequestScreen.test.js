import React from 'react'
import { shallow } from 'enzyme'
import { RideRequestScreen } from '../../../../../src/modules/rideRequest/containers/NewRideRequestContainer'
import { rideOfferFixture } from '../../../../resources/fixtures/ride/offer'

const props = {
  ride: rideOfferFixture,
  userId: 'kakaroto'
}

describe('<RideRequestScreen />', () => {
  const wrapper = shallow(<RideRequestScreen {...props} />)

  it('Snapshot have a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
