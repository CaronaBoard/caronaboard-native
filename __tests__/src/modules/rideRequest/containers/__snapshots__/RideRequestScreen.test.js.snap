// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RideRequestScreen /> Snapshot have a snapshot 1`] = `
<NewRideRequest
  ride={
    Object {
      "days": "Seg-Sex",
      "destination": "Bomfim",
      "driverId": "AIYmwTmrdTfUuGeuoNF0SYgXJ1j1",
      "hours": "19h",
      "origin": "PUC",
      "profile": Object {
        "contact": Object {
          "kind": "Whatsapp",
          "uid": "User ID",
          "value": "+5559999999",
        },
        "name": "Eduardo Moroni",
      },
      "rideId": "-KndyvnnlSN05mJxL57Q",
    }
  }
  saveRideRequest={[Function]}
/>
`;
